### DISCUSSION TITLE
Users with VRC+ custom profile picture

### DISCUSSION THREAD

**Discussion by Chayotea** [vrcx-team/VRCX#264] [General]
Seems if you use a custom profile picture you now appear in VRCX as default avatar image with unknown avatar. So far I find it funny that if you want some privacy on your current avatar selection you have to pay. If this was intentional then I guess now even in VRCX you can hide your current avatar selection by subscribing.

**Comment by Myrkie** [comment]
API updated anyone with vrcplus now has there avatar image forced to be the vrcplus image.

**Reply by Chayotea** [reply]
Main question is does this API update mean we can no longer see current avatar that is being used at all? I'm guessing if that is the case then I imagine VRCX updating with just VRC+ profile pics and "unknown avatar". 

At first I thought this was a thing they overlooked when they implemented the profile pic for VRC+ and I was hoping that in a update you could see both in a users profile and avatar picture in game. I guess with this API update the intention is clear and that is never going to happen.

**Reply by Myrkie** [reply]
"Unknown avatar" exists because of the way we grab the name of the avatar, if you remove /num/file from the end of a image url it provides the Json of the Image at upload which contains the avatars Name and uploader if there is no avatar image id we cannot grab it so it will come up as unknown.

**Reply by GenesisAria** [reply]
Not everyone with VRC+ shows unknown robot, many VRC+ people in my friends list still show their avatars.

**Reply by Myrkie** [reply]
to add onto this Anyone with vrcplus that has

userIcon
or
profilePicOverride

will have there currentAvatarImageUrl = https://assets.vrchat.com/system/defaultAvatar.png
aswell as 
currentAvatarThumbnailImageUrl = https://assets.vrchat.com/system/defaultAvatarThumbnail.png
since VRCX uses the thumbnail image of the current avatar to get the name it will always show as Unknown avatar

**Reply by GenesisAria** [reply]
I should be more specific, i have friends with VRC+ that have user icons, that also show their avatar.

**Reply by information-redacted** [reply]
> I should be more specific, i have friends with VRC+ that have user icons, that also show their avatar.

Does this happen within VRCX, or in-game?

**Reply by GenesisAria** [reply]
VRCX, ingame i wouldn't know how to check because it always shows user icon or profile image instead.

**Reply by Myrkie** [reply]
> I should be more specific, i have friends with VRC+ that have user icons, that also show their avatar.

This is caused because they dont have a 

profilePicOverride 

you can have a UserIcon but if you don't have a profile picture then it will continue to show your avatars image

**Reply by GenesisAria** [reply]
That was what i was guessing it was.  Thanks for clarifying.

**Reply by Myrkie** [reply]
> That was what i was guessing it was. Thanks for clarifying.

yep please refer to your original issue posted as  "fix" has been mentioned (enabling profilepicture override)

![image](https://user-images.githubusercontent.com/20288698/130079575-9662e744-71ea-4806-b0eb-77f4e34ac4dc.png)

**Reply by Chayotea** [reply]
Totally overlooked that option and enabled it immediately. I guess this will the default for now on.

**Reply by Myrkie** [reply]
unfortunately yes
